kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -223,6 +227,14 @@ class BatchJobSubmission(
               case Some(appId) => monitorBatchJob(appId)
             }
           case Some(metadata) =>
+            if (StringUtils.isNoneBlank(metadata.engineId, metadata.engineState)) {
+              _applicationInfo = Some(new ApplicationInfo(
+                id = metadata.engineId,
+                name = metadata.engineName,
+                state = ApplicationState.withName(metadata.engineState),
+                url = Option(metadata.engineUrl),
+                error = metadata.engineError))
```

Comment:
try to read the application info from metadata first.

In case it has been finished and metadata updated by peer.
https://github.com/apache/kyuubi/blob/7e199d6fdbdf52222bb3eadd056b9e5a2295f36e/kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala#L369-L385